sdv/vdb/client.py

```diff
@@ -41,16 +40,20 @@ class VehicleDataBrokerClient:
     def __new__(cls, port: Optional[int] = None):
         if cls._instance is None:
             cls._instance = super(VehicleDataBrokerClient, cls).__new__(cls)
-            if not conf.DISABLE_DAPR:
-                if port is None:
-                    port = int(str(os.getenv("DAPR_GRPC_PORT")))
-                cls._address = f"localhost:{port}"
+            service_locator = config.middleware.service_locator
+            _location = service_locator.get_service_location("vehicledatabroker")
+            _hostname = urlparse(_location).hostname
+            if port is None:
+                _port = urlparse(_location).port
             else:
-                cls._address = conf.VEHICLE_DATA_BROKER_ADDRESS
+                _port = port
+
+            _address = f"{_hostname}:{_port}"
```

Comment:
Should be checked if `_port` is `None`, otherwise we end up with "<hostname>:" ...

Reply:
As discussed, unnecessary check. That will be added to the documentation for the service name convention.